node/pkg/allocateip/allocateip_test.go

```diff
@@ -43,37 +44,13 @@ import (
 	"github.com/projectcalico/calico/libcalico-go/lib/options"
 )
 
-func allocateIPDescribe(description string, tunnelType []string, body func(tunnelType string)) bool {
```

Comment:
Looks like you opted to remove the custom describe and instead use a DescribeTable to get the same coverage - is that right? 

Could you elaborate on why that change is needed? Not necessarily against it, but would like to understand.

Reply:
I'm not 100% sure on the internals of ginkgo, but I was having trouble adding tests, and when I checked the logs, the custom describe implementation was not properly running all the scenarios for the different tunnelTypes (basically, it was running the same one repeatedly). I think that has to do with ginkgo's working in a 2-pass sort, and tables would make it work properly. Unfortunately I had to change a lot to make the test code more compatible with using tables...